jnosql-jakarta-persistence/jnosql-jakarta-persistence-driver/src/main/java/org/eclipse/jnosql/jakartapersistence/mapping/cache/PersistenceUnitCache.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *  The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ *  and the Apache License v2.0 is available at http://www.opensource.org/licenses/apache2.0.php.
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ *
+ *  Contributors:
+ *
+ *  Ondro Mihalyi
+ */
+package org.eclipse.jnosql.jakartapersistence.mapping.cache;
+
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.metamodel.EntityType;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Cache interface for persistence unit-level caching of metadata and queries.
+ * This cache is shared across all EntityManagers within the same EntityManagerFactory
+ * to improve performance by avoiding repeated computation of expensive operations.
+ *
+ * @author Ondro Mihalyi
+ */
+public interface PersistenceUnitCache {
+
+    /**
+     * Retrieves a map of entity types indexed by their simple names.
+     * Uses lazy initialization with double-checked locking pattern. The supplier must be set first, using the {@link #setEntityTypesByNameSupplier(java.util.function.Supplier) } method.
```

Comment:
PersistenceUnitCache.java:[36] (sizes) LineLength: Line is longer than 180 characters (found 186).

Reply:
Done